crates/test/macros/src/deassert.rs

```diff
@@ -0,0 +1,99 @@
+use quote::quote;
+use syn::{parse, Block, Expr, ExprMacro, ExprTuple, Stmt};
+
+/// Find and replace macro assertions inside the given block with `return Err(..)`.
+///
+/// The following assertion macros are replaced:
+/// - [`assert`]
+/// - [`assert_eq`]
+/// - [`assert_ne`]
```

Comment:
In addition to these macros, we should also try to hunt for and replace `.unwrap()` and `.expect(..)` with something that similarly causes `Err(..)` to be returned. That being said, I'm fine with this being best-effort; the goal is to make writing unit tests easy, not to completely erase knowledge of the fact that panics will break `cargo test`.

Reply:
That seems like a reasonable approach. Since we don't support `panic = "unwind"` (and [likely never will](https://github.com/flipperdevices/flipperzero-firmware/issues/1851)), I don't really see any other way of running tests than a) 1 binary per test, b) co-operative tests like you're doing here.